### DISCUSSION TITLE
how to use BOD on ATtiny1616

### DISCUSSION THREAD

**Discussion by cj190133** [SpenceKonde/megaTinyCore#669] [Q&A] [answered]
Hi guys,
for programming Attiny1616 I am using SerialUPDI programmer without any problem. (many thanks to SpenceKonde!) I don't use bootloader. But for using BOD functionality, there is notice in https://github.com/SpenceKonde/megaTinyCore, "You must do Burn Bootloader to apply this (BOD) setting".
Means it, I need burn the bootloader, because FUSES are programming only in bootloader burn process?
Means it also, can I (after bootloader is burned successfully) in next programming of Attiny1616, completely overwrite the bootloader?
Thank you in advance.
Best Regards

**Comment by SpenceKonde** [comment, marked as answer]
Yet another user getting confused by the poor naming of the "burn bootloader" menu option. God I hate the name of that menu option, it causes so many people to post questions just like this. I have tried to make this clear in the documentation, but apparently not clear enough.

"burn bootloader" should read "Set all fuses, and install bootloader (if any)". 
That describes what it does - all fuses are reset - either to what's specified by current menu selections, or the factory default. Additionally, in the event that the board definition specifies a bootloader file (which only the optiboot versions do)

But the Arduino IDE does not provide any facility for that text to be changed to accurately reflect what the option does; this is not specific to my cores. That's how it has always been on all AVR parts.

No bootloader is written when "burn bootloader" is chosen unless using an Optboot board definition (on non optiboot board definitions there is no bootloader to write!

The only difference is that when programming over UPDI, there are a small number of fuses that will always be written - we do this for fuses that are configured by menus (so something like the WDT fuse, we don't touch), but could not possibly brick the chip if set wrong (not th case for BOD - you can set the BOD threshold for a voltage higher than the power supply voltage. Depending on what you have soldered to the same board applying a high enough voltage to get past the BOD to reprogram it may require desoldering parts. Hence the BOD fuse is not a safe fuse, and is not one of the ones we set upon all uploads.

**Comment by cj190133** [comment]
Thank you SpenceKonde,  now it's clear.  I am going to use BOD with threshold 1.8V, therefore I shouldn't meet with problem when BOD threshold is higher as supply voltage. But good notice, it's easy to set BOD threshold to 4.2V by mistake ... it would be really problem for me, because Attiny is already soldered with another 3V devices ...
Thank you for clarification.

**Comment by SpenceKonde** [comment]
I think maybe part of the cause of the confusion was tat no core ever st fuuses on upload on classic
 AVRs because every fuse as at least oje bit that wuod brick the chip on most parts; 
biut over here the inquiries were cosant about sketches running at the wrong speed becase te had a 20 MHz-derved clock sleected but were running at 16 Mz  derived speed or the other way around.  And at that point I realized most fuses were safe
